tests/extra/test_job_management.py

```diff
@@ -455,7 +448,55 @@ def start_job(row, connection_provider, connection, **kwargs):
         assert set(result.status) == {"running"}
         assert set(result.backend_name) == {"foo"}
 
-
+    def test_cancel_prolonged_job_exceeds_duration(self):
+        # Create mock BatchJob instance
+        job = mock.MagicMock()  # Use MagicMock directly here
+        job.job_id = "test_job_id"
+        
+        row = {
+            "running_start_time": "2020-01-01T00:00:00Z"
+        }
+        
+        # Initialize manager with the max_running_duration as seconds
+        max_running_duration_seconds = 12 * 60 * 60  # 12 hours
+        manager = MultiBackendJobManager(cancel_running_job_after=max_running_duration_seconds)
+        
+        # set up timestamps
+        job_running_timestamp = rfc3339.parse_datetime(row["running_start_time"], with_timezone=True)
+        future_time = job_running_timestamp + datetime.timedelta(seconds=max_running_duration_seconds) + datetime.timedelta(seconds=1)
+
+        with time_machine.travel(future_time, tick=False):
+            manager._cancel_prolonged_job(job, row)
```

Comment:
I think this test should not call the private `_cancel_prolonged_job`. Running the public `run_jobs` would be ideal, but this involves an "infinite" while loop, so that is not ideal. 
I'd call at least `_update_statuses` to have better coverage of the actual flow

Reply:
struggling here to make it work. Given that you actually need to run the job to create an output csv. I was trying to access the job metapath but it also doesn't exist yet when I try to run my test.

Reply:
I believe I found a way!